cdap-data-fabric/src/main/java/co/cask/cdap/data2/nosql/NoSqlStructuredTable.java

```diff
@@ -114,6 +112,16 @@ public void close() throws IOException {
     table.close();
   }
 
+  private Optional<StructuredRow> read(Collection<Field<?>> keys, Collection<String> columns,
```

Comment:
I would find it more intuitive if this method takes columns=null to get all columns. Then read(keys, columns) should validate that the columns is not empty and call this. And convertKeysToBytes returns null for null.